controller/hybridgateway/target/target.go

```diff
@@ -0,0 +1,411 @@
+package target
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/samber/lo"
+	corev1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configurationv1alpha1 "github.com/kong/kong-operator/api/configuration/v1alpha1"
+	"github.com/kong/kong-operator/controller/hybridgateway/builder"
+	"github.com/kong/kong-operator/controller/hybridgateway/route"
+	"github.com/kong/kong-operator/controller/hybridgateway/utils"
+	gwtypes "github.com/kong/kong-operator/internal/types"
+)
+
+// ValidBackendRef represents a BackendRef that has passed all validation checks.
+type ValidBackendRef struct {
```

Comment:
ValidBackendRef is an internal type, used for the endpoint validation pipeline.
We use it as an internal type indeed.
We may want to not export it.
```suggestion
type validBackendRef struct {
```